web/tests/functional/store/test_store.py

```diff
@@ -0,0 +1,122 @@
+#
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""
+store tests.
+"""
+
+
+import json
+import os
+import plistlib
+import shutil
+import subprocess
+import unittest
+
+from libtest import env
+from libtest import plist_test
+
+from codechecker_common import util
+
+
+class TestStore(unittest.TestCase):
+    """Test storage reports"""
+
+    def setUp(self):
+
+        # Get the test workspace used to cppcheck tests.
+        self._test_workspace = os.environ["TEST_WORKSPACE"]
+
+        test_class = self.__class__.__name__
+        print("Running " + test_class + " tests in " + self._test_workspace)
+
+        self._test_cfg = env.import_test_cfg(self._test_workspace)
+
+    def test_tim_path_prefix_store(self):
+        """Trim the path prefix from the sored reports.
+
+        The source file paths are converted to absolute with the
+        temporary test directory, the test trims that temporary
+        test directory from the source file path during the storage.
+        """
+
+        test_dir = os.path.dirname(os.path.realpath(__file__))
+
+        report_dir = os.path.join(test_dir, "test_proj")
+
+        codechecker_cfg = self._test_cfg["codechecker_cfg"]
+
+        # Copy report files to a temporary directory not to modify the
+        # files in the repository.
+        # Report files will be overwritten during the tests.
+        temp_workspace = os.path.join(
+            codechecker_cfg["workspace"], "test_proj"
+        )
+        shutil.copytree(report_dir, temp_workspace)
+
+        report_file = os.path.join(temp_workspace, "divide_zero.plist")
+
+        # Convert file paths to absolute in the report.
+        plist_test.prefix_file_path(report_file, temp_workspace)
+
+        report_content = {}
+        with open(report_file, mode="rb") as rf:
+            report_content = plistlib.load(rf)
+
+        trimmed_paths = [
+            util.trim_path_prefixes(path, [temp_workspace])
+            for path in report_content["files"]
+        ]
+
+        run_name = "store_test"
+        store_cmd = [
+            env.codechecker_cmd(),
+            "store",
+            temp_workspace,
+            "--name",
+            run_name,
+            "--url",
+            env.parts_to_url(codechecker_cfg),
+            "--trim-path-prefix",
+            temp_workspace,
+            "--verbose",
+            "debug",
```

Comment:
It's easier to read if we format these lines in the following way:
```suggestion
            "--name", run_name,
            "--url", env.parts_to_url(codechecker_cfg),
            "--trim-path-prefix", temp_workspace,
            "--verbose", "debug",
```

Reply:
Probably for somebody it is easier to read that way, but I do not like to argue about formatting formatting that is why I've already suggested to introduce black in #2802 (with some configuration if needed), so we do not have to think about this stuff.
This format was generated with the default black so the benefit of this is that no manual work is needed to think about formatting and after a change it can be easily formatted again.